doc/sjs-for-js/es6-to-scala-part3.md

```diff
@@ -260,6 +260,234 @@ Here we use triple-quoted strings that allow us to write regex without escaping
+
+{% columns %}
+{% column 6 ES6 %}
+{% highlight javascript %}
+function compute(value, cPos, cNeg) {
+  if (value >=0)
+    return cPos();
+  else
+    return cNeg();
+}
+
+compute(x, () => expCalc(), () => expCalc2());
+{% endhighlight %}
+{% endcolumn %}
+
+{% column 6 Scala %}
+{% highlight scala %}
+def compute(value: Int, cPos: => Int, cNeg: => Int) = {
+  if (value >= 0)
+    cPos
+  else
+    cNeg
+}
+
+compute(x, expCalc, expCalc2)
+{% endhighlight %}
+{% endcolumn %}
+{% endcolumns %}
+
+#### Recursive functions
+
+Recursive functions can be very expressive, but they may also cause spurious stack overflows if the recursion gets too
+deep. Scala automatically optimizes recursive functions that are _tail recursive_, allowing you to use them without
+fear of overflowing the stack. To make sure your function is actually tail recursive, use the `@tailrec` annotation,
+which will cause the Scala compiler to report an error if your function is not tail recursive.
+
+JavaScript before ES6 did not support tail call optimization, nor optimizing tail recursive functions. If you use a
+smart ES6 transpiler, it can actually convert a tail recursive function into a `while` loop, but there are no checks
+available to help you to verify the validity of tail recursion.
+
+{% columns %}
+{% column 6 ES6 %}
+{% highlight javascript %}
+function fib(n){
+  function fibIter(n, a, b) {
+    if (n === 0) {
+      return b;
+    } else {
+      return fibIter(n - 1, a + b, a);
+    }
+  };
+  return fibIter(n, 1, 0);
+}
+{% endhighlight %}
+{% endcolumn %}
+
+{% column 6 Scala %}
+{% highlight scala %}
+def fib(n: Int): Int = {
+  @tailrec def fibIter(n: Int, a: Int, b: Int): Int = {
+    if (n == 0)
+      b
+    else
+      fibIter(n - 1, a + b, a)
+  }
+  fibIter(n, 1, 0)
+}
+{% endhighlight %}
+{% endcolumn %}
+{% endcolumns %}
+
+
+#### Partially applied functions
+
+In Scala you can call a function with only some of its arguments and get back a function taking those missing arguments.
+You do this by using `_` in place of the actual parameter. In JavaScript you can achieve the same by using the
+`Function.prototype.bind` function (although it limits you to providing parameters from left to right order). For
+example we can define a function to create HTML tags by wrapping content within start and end tags.
+
+{% columns %}
+{% column 6 ES6 %}
+{% highlight javascript %}
+function tag(name, content) {
+  return `<${name}>${content}</${name}>` 
+}
+
+const div = tag.bind(null, "div");
+const p = tag.bind(null, "p");
+const html = div(p("test")); // <div><p>test</p></div>
+{% endhighlight %}
+{% endcolumn %}
+
+{% column 6 Scala %}
+{% highlight scala %}
+def tag(name: String, content: String): String = {
+  s"<$name>$content</$name>"
+}
+
+val div = tag("div", _: String)
+val p = tag("p", _: String)
+val html = div(p("test")); // <div><p>test</p></div>
+{% endhighlight %}
+{% endcolumn %}
+{% endcolumns %}
+
+#### Multiple parameter lists
```

Comment:
I think this is too much. Or not well motivated.

Using multiple parameter lists is mostly useful if the last argument is a function type, and you want to call it with `{}` syntax. For example in `foldLeft`. Using it for currying is comparatively rare.